qsimcirq/qsim_simulator.py

```diff
@@ -391,8 +392,11 @@ def _sample_measure_results(
                 options["s"] = self.get_seed()
                 measurements = sampler_fn(options)
                 for key, bound in bounds.items():
+                    invert_mask = meas_ops[key].gate.full_invert_mask()
                     for j in range(bound[1] - bound[0]):
-                        results[key][i][j] = int(measurements[bound[0] + j])
+                        results[key][i][j] = int(
+                            measurements[bound[0] + j] ^ invert_mask[j]
+                        )
```

Comment:
Drive-by nit: doing this 3d loop in python, especially over the repetitions axis which could be very large, seems very inefficient. It'd be nice to change this to use numpy array operations to assign things in bulk as much as possible.

Reply:
Thanks for highlighting this - I'll open a follow-up PR to improve on it.